core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -315,6 +315,14 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     _dagScheduler = ds
   }
 
+  /**
+   * A unique identifier for the Spark application.
+   * Its format depends on the scheduler implementation.
+   * (i.e.
+   *  in case of local spark app something like u'local-1433865536131'
```

Comment:
Minor nit: the `u'` is going to be confusing to Scala users. I'd just go with regular quoted strings here.